src/app/Shared/Services/Settings.service.tsx

```diff
@@ -87,6 +95,28 @@ export class SettingsService {
     window.localStorage.setItem(StorageKeys.AutoRefreshUnits, String(units));
   }
 
+  automatedAnalysisRecordingConfig(
+    defaultConfig = defaultAutomatedAnalysisRecordingConfig
+  ): AutomatedAnalysisRecordingConfig {
+    const raw = window.localStorage.getItem(StorageKeys.AutomatedAnalysisRecordingConfig);
+    if (!!raw) {
+      try {
+        const config = JSON.parse(raw);
+        if (typeof config === 'object') {
+          return config;
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    }
+    this.setAutomatedAnalysisRecordingConfig(defaultConfig);
```

Comment:
I think we can remove the line `this.setAutomatedAnalysisRecordingConfig(defaultConfig);` since if no item is found in localstorage, we use the default anyways.